src/base64uploadadapter.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module upload/base64uploadadapter
+ */
+
+/* globals window */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import FileRepository from './filerepository';
+
+/**
+ * @extends module:core/plugin~Plugin
+ */
+export default class Base64UploadAdapter extends Plugin {
```

Comment:
Q: What if we named it simply `Base64Image`? The rest seems redundant. Then the 
* file name would be shorter,
* guide title would be shorter

Downside: worse SEO?

cc @Reinmar

Reply:
Good question. Especially that this thing doesn't actually upload (send) images. But if we'd go this way, then it should be moved out of `ckeditor5-upload` to... where? 

OTOH, if by "upload" we mean the whole process started by dropping an image, then this is an upload adapter. And it makes sense here, in this repo. And I'd call it consistently with other upload adapter classes.

Reply:
ATM:

```
src
├── base64uploadadapter.js
├── filereader.js
├── filerepository.js
└── ui
    └── filedialogbuttonview.js
```

Maybe let's reorganize things a little bit:

```
src
├── filereader.js
├── filerepository.js
├── adapters
│    ├── base64image.js
│    └── simpleimage.js
└── ui
    └── filedialogbuttonview.js
```

and rename classes to `Base64Image` and `SimpleImage`.

Reply:
```
├── adapters
│    ├── base64image.js
│    └── simpleimage.js
```

Even after moving those adapters to the dedicated namespace, I would like to end the class names with suffix "Adapter" because the `SimpleImage` class (even inside the namespace) does not say anything and can be easily used wrong.

Reply:
☝️ A good point, I guess.

Reply:
Could we move those adapters after closing this PR and before merging `SimpluUploadAdapter` (https://github.com/ckeditor/ckeditor5-upload/pull/94)?

Reply:
> Even after moving those adapters to the dedicated namespace, I would like to end the class names with suffix "Adapter" because the `SimpleImage` class (even inside the namespace) does not say anything and can be easily used wrong.

Big 👍 

Also, it makes it easier to find them. We had a couple of problems with classes with too short names (cause they were namespaced) and I stumbled upon this several times that it's hard to figure out what they are.